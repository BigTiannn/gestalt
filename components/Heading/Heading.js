import { createElement, PropTypes } from 'react';
import classnames from 'classnames/bind';
import styles from './Heading.css';

const cx = classnames.bind(styles);

const sizeElementMap = {
    xs: 'h5',
    s:  'h4',
    m: 'h3',
    l: 'h2',
    xl: 'h1'
};

export default function Heading(props) {
    const { children, color, size } = props;
    const cs = cx('Heading', `Heading--${size}`, `Heading--${color}`);
    return createElement(sizeElementMap[size], {className: cs}, children);
};

Heading.propTypes = {
    children: PropTypes.node.isRequired,
    color: PropTypes.oneOf(['white', 'lightgray', 'darkgray', 'blue', 'green']).isRequired,
    size: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl']).isRequired,
};

Heading.defaultProps = {
    color: 'darkgray'
};
