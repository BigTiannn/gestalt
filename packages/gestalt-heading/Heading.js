// @flow

import { createElement, Element, PropTypes } from 'react';
import classnames from 'classnames/bind';
import colorStyles from 'gestalt-colors/Colors.css';
import typographyStyles from 'gestalt-typography/Typography.css';
import whitespaceStyles from 'gestalt-whitespace/Whitespace.css';

const styles = {
  ...colorStyles,
  ...typographyStyles,
  ...whitespaceStyles,
};

const cx = classnames.bind(styles);

const sizesToElementMap = {
  xs: 'h5',
  s: 'h4',
  m: 'h3',
  l: 'h2',
  xl: 'h1',
};

type HeadingProps = {
  children?: Element<any>,
  color?: 'white' | 'gray' | 'dark-gray' | 'blue',
  size: 'xs' | 's' | 'm' | 'l' | 'xl',
  truncate?: bool,
};

export default function Heading(props: HeadingProps) {
  const {
    children,
    color = 'dark-gray',
    size,
    truncate = false,
  } = props;

  const cs = cx(
    'antialiased',
    'bold',
    'mb0',
    'mt0',
    'sans-serif',
    `display-${size}`,
    color,
    {
      truncate,
    },
  );
  return createElement(sizesToElementMap[size], { className: cs }, children);
}

Heading.propTypes = {
  children: PropTypes.node,
  color: PropTypes.oneOf(['white', 'gray', 'dark-gray', 'blue']),
  size: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl']).isRequired,
  truncate: PropTypes.bool,
};
