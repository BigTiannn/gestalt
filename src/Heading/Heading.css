.Heading {
    font-family: "Helvetica Neue";
    -webkit-font-smoothing: antialiased;

    margin-top: 0; /* .mt0 */
    margin-bottom: 0; /* .mb0 */
}

.Heading--xs {
    /* .text-8 */
    font-size: 24px;
    line-height: 28px;
    letter-spacing: -1px;
    word-spacing: -2px;
}

.Heading--s {
    /* .text-9 */
    font-size: 36px;
    line-height: 42px;
    letter-spacing: -1.25px;
    word-spacing: -2.25px;
}

.Heading--m {
    /* .text-10 */
    font-size: 48px;
    line-height: 56px;
    letter-spacing: -1.75px;
    word-spacing: -2.75px;
}

.Heading--l {
    /* .text-11 */
    font-size: 64px;
    line-height: 76px;
    letter-spacing: -2.25px;
    word-spacing: -3.25px;
}

.Heading--xl {
    /* .text-12 */
    font-size: 96px;
    line-height: 114px;
    letter-spacing: -3.25px;
    word-spacing: -4.25px;
}

.Heading--white {
    color: #fff; /* .white */
}

.Heading--lightgray {
    color: #b5b5b5;/* .gray */
}

.Heading--darkgray {
    color: #555; /* .black */
}

.Heading--blue {
    color: #0084ff; /* .blue */
}

.Heading--green {
    color: #88a5ab; /* .olive */
}
