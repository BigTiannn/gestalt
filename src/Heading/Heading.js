// @flow
import { createElement, PropTypes } from 'react';
import cx from 'classnames';
import colors from '../Colors.css';

const defaultHeadingLevels = {
  xs: 5,
  sm: 4,
  md: 3,
  lg: 2,
  xl: 1,
};

type Color = 'blue' | 'darkGray' | 'gray' | 'red' | 'white';

type Props = {|
  accessibilityLevel?: 1 | 2 | 3 | 4 | 5 | 6,
  children?: any,
  color?: Color,
  size: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
  truncate?: bool,
|};

export default function Heading(props: Props) {
  const {
    accessibilityLevel,
    children,
    color = 'darkGray',
    size,
    truncate = false,
  } = props;

  const cs = cx(
    'antialiased',
    'bold',
    'mb0',
    'mt0',
    'sans-serif',
    `display-${size}`,
    colors[color],
    'break-word',
    truncate,
  );

  const headingLevel = accessibilityLevel || defaultHeadingLevels[size];
  return createElement(`h${headingLevel}`, { className: cs }, children);
}

Heading.propTypes = {
  accessibilityLevel: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
  children: PropTypes.node,
  color: PropTypes.oneOf(['blue', 'darkGray', 'gray', 'red', 'white']),
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']).isRequired,
  truncate: PropTypes.bool,
};
