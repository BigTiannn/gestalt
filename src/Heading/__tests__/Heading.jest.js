/* eslint-env jest */
import React from 'react';
import { create } from 'react-test-renderer';
import Heading from '../Heading';

test('Heading small', () => {
  const tree = create(
    <Heading size="sm" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Heading xsmall with level 3', () => {
  const tree = create(
    <Heading size="xs" accessibilityLevel={3} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
