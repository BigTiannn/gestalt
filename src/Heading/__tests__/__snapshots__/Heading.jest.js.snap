exports[`test Heading small 1`] = `
<h4
  className="antialiased bold mb0 mt0 sans-serif display-sm darkGray break-word" />
`;

exports[`test Heading xsmall with level 3 1`] = `
<h3
  className="antialiased bold mb0 mt0 sans-serif display-xs darkGray break-word" />
`;
